# 카이제곱 검정
: 두 범주(카테고리)형 변수가 서로 상관이 있는지 통계정 검정방법
- 예) 학력,성별,직업의 만족도...


- 가설 = 시나리오
- 검정 = 테스트

#### 가설 : 아동별 장난감 보유 현황
- 귀무가설(니가 뭐라고 말하든 변하는거 없어!당연한거! 분석할 가치가 없는것)
>child 1 과 child 2의 장난감 보유수에 대한 만족도 차이가 없다.
- 대립가설(귀무가설과 완전 반대)
>child 1 과 child 2의 장난감 보유수에 대한 만족도 차이가 있다.
- 왜 차이가 나는지 이제 분석을 해보자!

In [2]:
#child 장난감 보유현황.
child1 <- c(5,11,1) 
child2 <-c(4,7,3)

In [3]:
#분석을 위해 데이터 합치기
Toy <- cbind(child1,child2)
Toy

child1 child2
[1,]  5     4     
[2,] 11     7     
[3,]  1     3

In [8]:
#장난감 이름 정해주기
rownames(Toy) <- c("car","truck", "doll")
Toy

child1 child2
car    5     4     
truck 11     7     
doll   1     3

In [10]:
#카이제곱 검정 실시
chisq.test(Toy)

# 가장 중요한건 : p-value 유의수준?

Warning message in chisq.test(Toy):
“Chi-squared approximation may be incorrect”


	Pearson's Chi-squared test

data:  Toy
X-squared = 1.7258, df = 2, p-value = 0.4219


### 결론
-  p-value = 0.4219 > 0.05(95% 신뢰구간) -> 귀무가설 채택
> child1과 child2의 데이터는 통계적으로 큰 차이가 없다.
> 저 데이터가지고는 분석할 거리가 안된다!
> 
-  p-value는 0에 가까울수록 좋다.
> 미리 정해진 유의수준(일반적으로 R에서는 0.05)보다 작으면 대립가설을 채택하고 유의수준보다 크면 귀무가설을 채택.

In [12]:
cor(child1,child2)

[1] 0.9862414

### 메세지
- “Chi-squared approximation may be incorrect”
> 카이제곱의 근사값이 정확하지 않을수도 있다.
- 데이터 양이 적은경우 발생

---
# 피셔 검정
> 데이터 양이 적을경우 사용


- 원래는 카이제곱을 더 많이 사용

In [16]:
fisher.test(Toy)


	Fisher's Exact Test for Count Data

data:  Toy
p-value = 0.5165
alternative hypothesis: two.sided


p-value = 0.5165 > 0.05 --> 귀무가설

---
# t-검정
- 가설 : 건전지가 1000시간 간다고 했는데 그거보다 짧게 수명이 다되었다!
> 다른사람이 한 말이 진짜인이 아닌지를 볼때 사용(평균!)


- 귀무가설 (분석할 필요가 없다) 
> 건전지 수명은 1000시간이다.
- 대립가설
> 건전지 수명은 1000시간이 아니다.

In [19]:
bat <- c(980, 1008, 968, 1032, 1012, 1002, 996, 1017)
# 진짜인지 아닌지 확인 : 정규 문포도

- 데이터의 분포가 정규분포인지 확인 --> Shapiro-Wilk검정을 실시
- 귀무가설 : 자료가 정규분포를 따른다.
- 대립가설 : 자료가 정규분포를 따르지 않는다.


In [20]:
shapiro.test(bat)


	Shapiro-Wilk normality test

data:  bat
W = 0.97706, p-value = 0.9469


p-value = 0.9469 > 0.05 ---> 귀무가설

In [22]:
#t검정 실시
t.test(bat, mu=1000, alternative = "two.sided")

#mu : 비교하는 대상의 평균
#alternative :
#- two.sided : 데이터가 평균과 다르다.
#- greater : 데이터가 평균보다 크다.
#- less : 데이터가 평균보다 적다.


	One Sample t-test

data:  bat
t = 0.25891, df = 7, p-value = 0.8032
alternative hypothesis: true mean is not equal to 1000
95 percent confidence interval:
  984.7508 1018.9992
sample estimates:
mean of x 
 1001.875 


p-value = 0.8032 > 0.05 ---> 귀무가설 : 건전지의 수명은 1000시간이다.